students/g_rama/lesson06/test_mailroom_v4.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+from unittest import TestCase, mock
+import pytest
+import mailroom_v4
+import unittest
+from io import StringIO
+from testfixtures import tempdir, compare
+import os
+
+
+@pytest.mark.parametrize('name, amount, expected', [
+    ("dan", '50', "Thank you dan for donating 50 dollars generously."),
+    ("jeff", '60', "Thank you jeff for donating 60 dollars generously.")
+])
+def test_thank_you_letter_positive(name, amount, expected):
+    result = str(mailroom_v4.thank_you_letter(name, amount))
+    assert expected == result
+
+
+@pytest.mark.parametrize('name, amount, expected', [
+    ("dan", 50, "Thank you sam for donating 50 dollars generously."),
+    ("jeff", 60, "Thank you  for donating 60 dollars generously.")
+])
+def test_thank_you_letter_negitive(name, amount, expected):
+    result = str(mailroom_v4.thank_you_letter(name, amount))
+    assert expected != result
+
+
+testing_donors_data = {"testname1": [200, 20, 35.5],
+                   "testname2": [500, 20],
+                   "Susan": [1000, 20, 70],
+                   "Rob": [250, 20],
+                   }
+
+
+@unittest.mock.patch('mailroom_v4.donor_details')
+def test_donor_details(mock_donor_details):
+    mailroom_v4.donor_details(testing_donors_data)
+    mock_donor_details.assert_called_with(testing_donors_data)
+
+
+def test_amount_validate_positive():
+    assert mailroom_v4.amount_validate(float(20))
+
+
+@unittest.mock.patch('mailroom_v4.amount_validate')
+def test_amount_validate_negitive(mock_amount_validate):
+    mailroom_v4.amount_validate(-10)
```

Comment:
same comment here - you're testing the mock.

Reply:
Kind of confused when you say "testing the mock" Although function is mocked, I am calling the  function from mailroom_v4? doesn't it mean I will be testing the functionality of that function?